sklearn/linear_model/least_angle.py

```diff
@@ -1468,8 +1469,9 @@ def fit(self, X, y, copy_X=True):
             df[k] = np.sum(mask)
 
         self.alphas_ = alphas_
-        with np.errstate(divide='ignore'):
-            self.criterion_ = n_samples * np.log(mean_squared_error) + K * df
+        eps64 = np.finfo('float64').eps
+        self.criterion_ = (n_samples * mean_squared_error / (sigma2 + eps64) +
```

Comment:
OK, it took me a while to understand that the reason that we don't need the log is that the MSE is actually already the log-likelihood.